### PR TITLE
Add open exchage rates API key

### PR DESCRIPTION
**Pull request by MatthewPiatetsky**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [x] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [x] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [x] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -400,3 +400,6 @@
 - Role: edxapp
   - Added `EDXAPP_VIDEO_TRANSCRIPTS_SETTINGS` to configure S3-backed video transcripts.
   - Removed unused `EDXAPP_BOOK_URL` setting
+
```
Comment (feanil) [inline CHANGELOG.md]: Please add this change to the top of the file.  People adding it to the bottom is incorrect we just don't always catch it.

File: playbooks/roles/discovery/defaults/main.yml [hunk 2]
```diff
@@ -123,5 +123,7 @@ discovery_service_config_overrides:
 
   PUBLISHER_FROM_EMAIL: '{{ DISCOVERY_PUBLISHER_FROM_EMAIL }}'
 
+  OPENEXCHANGERATES_API_KEY: '{{ OPENEXCHANGERATES_API_KEY }}'
```
Comment (feanil) [inline playbooks/roles/discovery/defaults/main.yml]: The toplevel var `OPENEXCHANGERATES_API_KEY` is never defined.  You should define a var called `DISCOVERY_OPENEXCHANGERATES_API_KEY` outside this dictionary and then reference it here.  The var that we would set in the secure config is then `DISCOVERY_OPENEXCHANGERATES_API_KEY`.

You can find more on ansible variables here: http://docs.ansible.com/ansible/latest/playbooks_variables.html

File: CHANGELOG.md (hunk 1 above)
Comment (MatthewPiatetsky) [reply CHANGELOG.md]: i see, example PR i saw had it the other way. changed

File: playbooks/roles/discovery/defaults/main.yml (hunk 2 above)
Comment (MatthewPiatetsky) [reply playbooks/roles/discovery/defaults/main.yml]: I see, made default ''

**Review by feanil — COMMENTED** [review]
@MatthewPiatetsky  looks like your rebase picked up a commit that is not yours.  Can you fix that?  After that, this looks good.

**Review by jdmulloy — APPROVED** [review]

**Review by feanil — APPROVED** [review]

**Comment by MatthewPiatetsky** [comment]
@jdmulloy let me know when the secure config pr is done and then i can merge this one

**Comment by feanil** [comment]
@MatthewPiatetsky you can merge this one without waiting for that one, since you have defaults here. It shouldn't break anything.

**Comment by MatthewPiatetsky** [comment]
ok, i will merge this one, but i still need to hold off merging the course-discovery and edx-mktg PRs

**Pull request merged**